src/panels/lovelace/views/hui-view.ts

```diff
@@ -86,6 +90,23 @@ export class HUIView extends LitElement {
     return element;
   }
 
+  public createBadgeElement(badgeConfig: LovelaceBadgeConfig) {
+    const element = createBadgeElement(badgeConfig) as LovelaceBadge;
+    element.hass = this.hass;
+    element.addEventListener(
```

Comment:
This event listener is now only attached at a rebuild, so it will never be attached, because it doesn't listen for a rebuild.

Reply:
the event listener is attached when creating the all badges

```js
 private _createBadges(config: LovelaceViewConfig): void {
...
    for (const badge of badges) {
      const element = createBadgeElement(badge);
...
```

Reply:
OK, then you are just assigning `hass` twice, remove it from `_createBadges` you are doing it in `createBadgeElement`

Reply:
I don't know why, but setting it only in `createBadgeElement` makes the badges take a really long time to render